--- uprof/Makefile.am
+++ uprof/Makefile.am
@@ -53,16 +53,6 @@ libuprof_@UPROF_MAJOR_VERSION@_@UPROF_MINOR_VERSION@_la_CPPFLAGS = \
     	-I$(top_srcdir)/uprof \
     	@EXTRA_CPPFLAGS@
 
-if HAVE_NCURSES
-
-bin_PROGRAMS = uprof-tool
-uprof_tool_SOURCES = uprof-tool.c uprof-marshal.c
-uprof_tool_LDFLAGS = @UPROF_DEP_LIBS@ -luprof-@UPROF_MAJOR_VERSION@.@UPROF_MINOR_VERSION@ -lncursesw
-uprof_tool_CFLAGS = @EXTRA_CFLAGS@ @UPROF_DEP_CFLAGS@
-uprof_tool_CPPFLAGS = @EXTRA_CPPFLAGS@
-
-endif
-
 uprof-service-glue.h: org.freedesktop.UProf.Service.xml Makefile.am
 	$(AM_V_GEN)dbus-binding-tool --prefix=_uprof_service --mode=glib-server --output=uprof-service-glue.h $(srcdir)/org.freedesktop.UProf.Service.xml
 
