diff --git c/clutter/clutter-actor.c w/clutter/clutter-actor.c
index 0fc8df6..21a2572 100644
--- clutter/clutter-actor.c
+++ clutter/clutter-actor.c
@@ -13292,6 +13292,7 @@ clutter_actor_set_child_above_sibling (ClutterActor *self,
                                     ADD_CHILD_NOTIFY_FIRST_LAST,
                                     insert_child_above,
                                     sibling);
+  g_object_unref(child);
 
   clutter_actor_queue_relayout (self);
 }
@@ -13338,6 +13339,7 @@ clutter_actor_set_child_below_sibling (ClutterActor *self,
                                     ADD_CHILD_NOTIFY_FIRST_LAST,
                                     insert_child_below,
                                     sibling);
+  g_object_unref(child);
 
   clutter_actor_queue_relayout (self);
 }
@@ -13376,6 +13378,7 @@ clutter_actor_set_child_at_index (ClutterActor *self,
                                     ADD_CHILD_NOTIFY_FIRST_LAST,
                                     insert_child_at_index,
                                     GINT_TO_POINTER (index_));
+  g_object_unref (child);
 
   clutter_actor_queue_relayout (self);
 }
